modules/http/http-curl-compression.h

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (c) 2022 One Identity LLC.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * As an additional exemption you are allowed to compile & link against the
+ * OpenSSL libraries as published by the OpenSSL project. See the file
+ * COPYING for details.
+ *
+ */
+
+#ifndef SYSLOG_NG_HTTP_CURL_COMPRESSION_H
+#define SYSLOG_NG_HTTP_CURL_COMPRESSION_H
+
+/* Uncomment upon implementation */
+#define CURL_COMPRESSION_IMPLEMENTED_GZIP
+#define CURL_COMPRESSION_IMPLEMENTED_DEFLATE
+//#define CURL_COMPRESSION_IMPLEMENTED_BROTLI
+//#define CURL_COMPRESSION_IMPLEMENTED_ZSTD
+
+#include <glib.h>
+
+enum CurlCompressionTypes
+{
+  CURL_COMPRESSION_UNCOMPRESSED = 0,
+  CURL_COMPRESSION_GZIP = 1,
+  CURL_COMPRESSION_DEFLATE = 2,
+  CURL_COMPRESSION_BROTLI = 3,
+  CURL_COMPRESSION_ZSTD = 4
+};
+extern gint8 CURL_COMPRESSION_TYPES_LEN;
+
+extern gint8 CURL_COMPRESSION_DEFAULT;
+extern gchar *CURL_COMPRESSION_LITERAL_ALL;
+
+extern gchar *curl_compression_types[];
+gboolean http_dd_compress_string(GString *compression_destination, const GString *message, const gint compression);
```

Comment:
Passing the `compression` in is not very OOP-style. I could imagine a general `Compression` interface, which would be implemented by `GZipCompression`, `DeflateCompression`, `BrotliCompression`, etc...
The user could call `brotli_compression_compress()` for example.

Reply:
Thait is true, however it fits with C conventions, and does not violate separation of concerns IMHO. If you insist, I could refactor, but I'd rather not.

Reply:
We are trying to live up to 21st century C standards, doing everything we can in OOP style.
Clean coding and maintainability are really important to the project, and if we can implement the same functionality with more modern philosophy, I think we should.

It is not up to me to decide where we draw the line, though. @MrAnno @bazsi What do you think?

Reply:
Done.